pam/integration-tests/cmd/exec-client/modulewrapper.go

```diff
@@ -28,3 +32,30 @@ func (m *moduleWrapper) SimulateClientPanic(text string) {
 func (m *moduleWrapper) SimulateClientError(errorMsg string) error {
 	return errors.New(errorMsg)
 }
+
+// SimulateClientSignal sends a signal to the child process.
+func (m *moduleWrapper) SimulateClientSignal(sig syscall.Signal) {
+	pid := os.Getpid()
+	log.Debugf(context.Background(), "Sending signal %v to self pid (%v)",
+		sig, pid)
+
+	c := make(chan os.Signal, 1)
+	done := make(chan struct{})
+	signal.Notify(c, os.Interrupt)
```

Comment:
This only relays the `os.Interrupt` signal to `c`. I think you want to use just `signal.Notify(c)` here.

Reply:
or `signal.Notify(c, sig)`, but I'm not sure if you can just use a `syscall.Signal` as an `os.Signal`

Reply:
Yeah `syscall.Signal` implements the `os.Signal` iface.

And what you say is true, here in fact we didn't really want to handle those signals otherwise we'd never actually handle them in the lower stack, so l changed the code to do what we actually want in case the signal is expected to terminate the program.